aep/general/0161/aep.md.j2

```diff
@@ -1,5 +1,135 @@
+
+```proto
+message UpdateBookRequest {
+  // The book to update.
+  //
+  // The book's `name` field is used to identify the book to update.
+  // Format: publishers/{publisher}/books/{book}
+  Book book = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The list of fields to update.
+  // Fields are specified relative to the book
+  // (e.g. `title`, `rating`; *not* `book.title` or `book.rating`).
+  google.protobuf.FieldMask update_mask = 2;
+}
+```
+
+### Read-write consistency
+
+Read and write behavior for field masks **must** be self-consistent if a mask
+is present:
+
+- If a user updates a resource with a given mask, and then reads the same
+  resource with the same mask, the exact same data **must** be returned.
+  - Exception: [Output only fields](#output-only-fields).
+- Similarly, reading a resource with a given mask and then updating the
+  resource with the returned data and the same mask **must** be a no-op.
+
+**Note:** This implies that any mask that is valid for either read or write
+**must** be valid for both.
+
+### Specifying specific fields
+
+Field masks **must** permit the specification of specific fields in a defined
+struct, using the `.` character for traversal.
+
+Because field masks are always relative to the resource, direct fields on the
+resource require no traversal (examples: `title`, `rating`). Traversal is used
+when resources contain messages (example: `author.given_name`).
+
+**Note:** A user **must** be able to specify either a field as a whole, or one
+of its subfields: `author` and `author.given_name` are both valid.
+
+### Map fields
+
+Field masks **may** permit the specification of specific fields in a map, if
+and only if the map's keys are either strings or integers, using the `.`
+character for traversal.
+
+Field masks **should** support string keys that contain characters that are
+problematic for the field mask syntax, using the backtick character.
+
+```proto
+message Book {
+  // The name of the book.
+  // Format: publishers/{publisher}/books/{book}
+  string name = 1;
+
+  // Reviews for the back cover. The key is the author of the review,
+  // and the value is the text of the review.
+  //
+  // Valid field masks: reviews, reviews.smith, reviews.`John Smith`
+  map<string, string> reviews = 2;
+}
+```
+
+### Wildcards
+
+Field masks **may** permit the use of the `*` character on a repeated field or
+map to indicate the specification of particular sub-fields in the collection:
+
+```proto
+message Book {
+  option (google.api.resource) = {
+    type: "library.googleapis.com/Book"
+    pattern: "publishers/{publisher}/books/{book}"
+  };
+
+  // The name of the book.
+  // Format: publishers/{publisher}/books/{book}
+  string name = 1 [(google.api.field_behavior) = IDENTIFIER];
+
+  // The author or authors of the book.
+  // Valid field masks: authors, authors.*.given_name, authors.*.family_name
+  // Invalid field masks: authors.0, authors.0.given_name
```

Comment:
Can you also add 'authors[0]' style examples, since those might be intuitive for users?